### PR TITLE
test(item): form data for items with controls

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Tests

## What is the current behavior?

The "item: inputs" test suite times out often within Playwright & CI. We are currently skipping those tests in `main`. 

Issue URL: Internal

## What is the new behavior?

- Removes excessive conditions from test suite
- Splits up test to verify independent pieces of functionality 
- Removed extraneous HTML from the test template

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/item/test/inputs/item.e2e.ts [hunk 1]
```diff
@@ -1,118 +1,137 @@
...
+
+      await expect(await formResult.textContent()).toEqual(
+        '{"input":"","textarea":"","toggle":"","checkbox":"","select":"","datetime":"2022-04-01T10:00","range":"10"}'
+      );
     });
 
-    // Reenable and set some value
-    await disableToggle.click();
-    await page.click('#btnSomeValue');
+    test('form controls have some value', async ({ page }) => {
```
Comment (sean-perkins) [inline core/src/components/item/test/inputs/item.e2e.ts:52]: Need to skip all non-screenshot tests on RTL.

**Review by amandaejohnston — APPROVED** [review]

File: packages/vue/src/vue-component-lib/utils.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { VNode, defineComponent, getCurrentInstance, h, inject, ref, Ref } from 'vue';
 
-export interface InputProps {
-  modelValue?: string | boolean;
+export interface InputProps<T> {
```
Comment (liamdebeasi) [inline packages/vue/src/vue-component-lib/utils.ts:3]: Can we remove this? Looks like the vue output target changes got into this PR.

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: packages/vue/src/vue-component-lib/utils.ts (hunk 2 above)
Comment (sean-perkins) [reply packages/vue/src/vue-component-lib/utils.ts:3]: Ahh yep, didn't re-install our lock version after experimenting with a dev-build. I'll revert this change.

**Review by liamdebeasi — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**